Source/Graphic/HCPath+SVG.c

```diff
@@ -0,0 +1,409 @@
+        angleSpan = fabs(angleSpan - 2.0 * M_PI);
+    }
+    
+    // Calculate cubic curves to represent the arc, splitting the angle span when it exceeds the defined maximum (typically pi/2)
+    HCReal angleSpanRemaining = angleSpan;
+    HCReal angleSpanMaximum = M_PI * 0.5;
+    HCReal angle = angleStart;
+    while (angleSpanRemaining > 0.0) {
+        // Calculate the angle slice to be represented by this curve
+        HCReal angleSliceSpan = fmin(angleSpanRemaining, angleSpanMaximum);
+        HCReal angleSlice = angleSliceSpan * (sweep ? 1.0 : -1.0);
+        HCReal angleSliceEnd = fmod(angle + angleSlice , 2.0 * M_PI);
+        
+        // Calculate a representation of the angle slice on the unit circle using a cubic curve
+        HCReal approximation = 4.0 / 3.0 * tan(angleSlice * 0.25);
+        HCReal rotatedCosPhi = cos(angleSlice);
+        HCReal rotatedSinPhi = sin(angleSlice);
+        HCReal cx0Rotated = 1.0;
+        HCReal cy0Rotated = approximation;
+        HCReal cx1Rotated = rotatedCosPhi + approximation * rotatedSinPhi;
+        HCReal cy1Rotated = rotatedSinPhi - approximation * rotatedCosPhi;
+        HCReal xRotated = rotatedCosPhi;
+        HCReal yRotated = rotatedSinPhi;
+
+        // Rotate and scale the unit circle approximation curve to fix the desired ellipse size and eccentricity
+        HCReal cosAngle = cos(angle);
+        HCReal sinAngle = sin(angle);
+        HCReal cx0p = xr * (cosAngle * cx0Rotated - sinAngle * cy0Rotated);
+        HCReal cy0p = yr * (sinAngle * cx0Rotated + cosAngle * cy0Rotated);
+        HCReal cx1p = xr * (cosAngle * cx1Rotated - sinAngle * cy1Rotated);
+        HCReal cy1p = yr * (sinAngle * cx1Rotated + cosAngle * cy1Rotated);
+        HCReal px1p = xr * (cosAngle * xRotated - sinAngle * yRotated);
+        HCReal py1p = yr * (sinAngle * xRotated + cosAngle * yRotated);
+        
+        // Rotate the ellipse by the desired rotation angle
+        HCReal cosRotation = cos(rotation);
+        HCReal sinRotation = sin(rotation);
+        HCReal cx0 = cosRotation * cx0p - sinRotation * cy0p + cx;
+        HCReal cy0 = sinRotation * cx0p + cosRotation * cy0p + cy;
+        HCReal cx1 = cosRotation * cx1p - sinRotation * cy1p + cx;
+        HCReal cy1 = sinRotation * cx1p + cosRotation * cy1p + cy;
+        HCReal px1 = cosRotation * px1p - sinRotation * py1p + cx;
+        HCReal py1 = sinRotation * px1p + cosRotation * py1p + cy;
+
+        // Determine how much more of the arc needs to be approximated
+        angleSpanRemaining -= angleSliceSpan;
+        angle = angleSliceEnd;
+        
+        // Where this is the last slice of the arc, ensure the final point is exactly as requested
+        if (angleSpanRemaining <= 0.0) {
+            px1 = x;
+            py1 = y;
+        }
+        
+        // Add the completed approximation curve for the angle slice to the path
+        HCPathAddCubicCurve(self, cx0, cy0, cx1, cy1, px1, py1);
+        
+#define VISUALIZE_ARC false
```

Comment:
Just checking if you want this `#define` here instead of the top of the file? At the top it would be easier to enable/disable.

Reply:
I didn't plan on it being file-level or object-level. Its just here for anyone trying to determine how this function works. I made it a #define to avoid performance costs checking  a variable.